diff --git a/boot/isolinux/isolinux.cfg b/boot/isolinux/isolinux.cfg
index 01ff90e..3fa12e8 100644
--- a/boot/isolinux/isolinux.cfg
+++ b/boot/isolinux/isolinux.cfg
@@ -23,7 +23,7 @@ label debug
 
 label install
 	menu label ^Installation - Install OS_TITLE to harddisk
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE INSTALL=1 DEBUG=
 
 menu separator
@@ -41,12 +41,12 @@ menu separator
 
 label auto_install
 	menu label Audo_^Installation - Auto Install to specified harddisk
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE AUTO_INSTALL=0 DEBUG=
 
 label auto_update
 	menu label Audo_^Update - Auto update OS_TITLE
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE AUTO_INSTALL=update DEBUG=
 
 menu separator
diff --git a/install/grub2/efi/boot/android.cfg b/install/grub2/efi/boot/android.cfg
index f26dcbc..8bb42a6 100644
--- a/install/grub2/efi/boot/android.cfg
+++ b/install/grub2/efi/boot/android.cfg
@@ -1,11 +1,12 @@
 # $1 Title
-# $2... Kernel cmdline
+# $2 Kernel image
+# $3... Kernel cmdline
 function add_entry {
 	menuentry "OS_TITLE VER $1" "$@" --class android-x86 {
 		shift 2
 		savedefault
 		set root=$android
-		linux $kdir/kernel CMDLINE $src $@
+		linux $kdir/$2 CMDLINE $src $@
 		initrd $kdir/initrd.img
 	}
 }
@@ -50,17 +51,17 @@ search --no-floppy --set android -f $kdir/kernel
 export android bootefi grub kdir live src
 
 # Create main menu
-add_entry "$live" quiet
-add_entry "$debug_mode" DEBUG=2
+add_entry "$live" kernel quiet
+add_entry "$debug_mode" kernel DEBUG=2
 if [ -s ($android)$kdir/install.img ]; then
-	add_entry "Installation" INSTALL=1
-fi
+	add_entry "Installation" install INSTALL=1
+fi/
 
 submenu "Advanced options -> " {
-	add_entry "$live - No Hardware Acceleration" quiet nomodeset HWACCEL=0
+	add_entry "$live - No Hardware Acceleration" kernel quiet nomodeset HWACCEL=0
 	if [ -s ($android)$kdir/install.img ]; then
-		add_entry "Auto Install to specified harddisk" AUTO_INSTALL=0
-		add_entry "Auto Update" AUTO_INSTALL=update
+		add_entry "Auto Install to specified harddisk" install AUTO_INSTALL=0
+		add_entry "Auto Update" install AUTO_INSTALL=update
 	fi
 	add_os_if_exists /EFI/BOOT/$bootefi "UEFI OS"
 	add_os_if_exists /EFI/BOOT/fallback.efi "UEFI Fallback"
