diff --git a/boot/isolinux/isolinux.cfg b/boot/isolinux/isolinux.cfg
index 01ff90e..3fa12e8 100644
--- a/boot/isolinux/isolinux.cfg
+++ b/boot/isolinux/isolinux.cfg
@@ -23,7 +23,7 @@ label debug
 
 label install
 	menu label ^Installation - Install OS_TITLE to harddisk
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE INSTALL=1 DEBUG=
 
 menu separator
@@ -41,12 +41,12 @@ menu separator
 
 label auto_install
 	menu label Audo_^Installation - Auto Install to specified harddisk
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE AUTO_INSTALL=0 DEBUG=
 
 label auto_update
 	menu label Audo_^Update - Auto update OS_TITLE
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE AUTO_INSTALL=update DEBUG=
 
 menu separator
