diff --git a/boot/isolinux/isolinux.cfg b/boot/isolinux/isolinux.cfg
index 01ff90e..3fa12e8 100644
--- a/boot/isolinux/isolinux.cfg
+++ b/boot/isolinux/isolinux.cfg
@@ -23,7 +23,7 @@ label debug
 
 label install
 	menu label ^Installation - Install OS_TITLE to harddisk
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE INSTALL=1 DEBUG=
 
 menu separator
@@ -41,12 +41,12 @@ menu separator
 
 label auto_install
 	menu label Audo_^Installation - Auto Install to specified harddisk
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE AUTO_INSTALL=0 DEBUG=
 
 label auto_update
 	menu label Audo_^Update - Auto update OS_TITLE
-	kernel /kernel
+	kernel /install
 	append initrd=/initrd.img CMDLINE AUTO_INSTALL=update DEBUG=
 
 menu separator
diff --git a/install/grub2/efi/boot/android.cfg b/install/grub2/efi/boot/android.cfg
index f26dcbc..fd25da0 100644
--- a/install/grub2/efi/boot/android.cfg
+++ b/install/grub2/efi/boot/android.cfg
@@ -10,6 +10,18 @@ function add_entry {
 	}
 }
 
+# $1 Title
+# $2... kernel cmdline
+function add_entry_vesa {
+	menuentry "OS_TITLE VER $1" "$@" --class android-x86 {
+		shift 2
+		savedefault
+		set root=$android
+		linux $kdir/install CMDLINE $src $@
+		initrd $kdir/initrd.img
+	}
+}
+
 # $1 EFI to chainload
 # $2 OS name
 # $3 Class
@@ -53,14 +65,14 @@ export android bootefi grub kdir live src
 add_entry "$live" quiet
 add_entry "$debug_mode" DEBUG=2
 if [ -s ($android)$kdir/install.img ]; then
-	add_entry "Installation" INSTALL=1
+	add_entry_vesa "Installation" INSTALL=1
 fi
 
 submenu "Advanced options -> " {
 	add_entry "$live - No Hardware Acceleration" quiet nomodeset HWACCEL=0
 	if [ -s ($android)$kdir/install.img ]; then
-		add_entry "Auto Install to specified harddisk" AUTO_INSTALL=0
-		add_entry "Auto Update" AUTO_INSTALL=update
+		add_entry_vesa "Auto Install to specified harddisk" AUTO_INSTALL=0
+		add_entry_vesa "Auto Update" AUTO_INSTALL=update
 	fi
 	add_os_if_exists /EFI/BOOT/$bootefi "UEFI OS"
 	add_os_if_exists /EFI/BOOT/fallback.efi "UEFI Fallback"
