diff --git a/core/main.mk b/core/main.mk
index f5baeac..bbbdb2a 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -315,14 +315,6 @@ include build/core/pdk_config.mk
 # Jacoco package name for Jack
 -include $(TOPDIR)external/jacoco/config.mk
 
-#
-# -----------------------------------------------------------------
-# Enable dynamic linker developer warnings for all builds except
-# final release.
-ifneq ($(PLATFORM_VERSION_CODENAME),REL)
-  ADDITIONAL_BUILD_PROPERTIES += ro.bionic.ld.warning=1
-endif
-
 # -----------------------------------------------------------------
 ###
 ### In this section we set up the things that are different
