diff --git a/core/main.mk b/core/main.mk
index 3d42c8f72..29784399c 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -185,19 +185,6 @@ include build/core/pdk_config.mk
 # Jacoco package name for Jack
 -include $(TOPDIR)external/jacoco/config.mk
 
-#
-# -----------------------------------------------------------------
-# Enable dynamic linker developer warnings for userdebug, eng
-# and non-REL builds
-ifneq ($(TARGET_BUILD_VARIANT),user)
-  ADDITIONAL_BUILD_PROPERTIES += ro.bionic.ld.warning=1
-else
-# Enable it for user builds as long as they are not final.
-ifneq ($(PLATFORM_VERSION_CODENAME),REL)
-  ADDITIONAL_BUILD_PROPERTIES += ro.bionic.ld.warning=1
-endif
-endif
-
 ADDITIONAL_BUILD_PROPERTIES += ro.treble.enabled=${PRODUCT_FULL_TREBLE}
 
 # -----------------------------------------------------------------
